pkg/download/catalog.go

```diff
@@ -0,0 +1,66 @@
+package download
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/gomods/athens/pkg/paths"
+
+	"github.com/gobuffalo/buffalo"
+	"github.com/gobuffalo/buffalo/render"
+	"github.com/gomods/athens/pkg/errors"
+	"github.com/gomods/athens/pkg/log"
+)
+
+// PathCatalog URL.
+const PathCatalog = "/catalog"
+const DefaultPageSize = 1000
+
+type catalogRes struct {
+	ModsAndVersions []paths.AllPathParams
+	NextPageToken   string
+}
+
+// CatalogHandler implements GET baseURL/catalog
+func CatalogHandler(dp Protocol, lggr log.Entry, eng *render.Engine) buffalo.Handler {
+	const op errors.Op = "download.CatalogHandler"
+
+	return func(c buffalo.Context) error {
```

Comment:
Question: Can it handle normal context.Context?

Reply:
Not sure I understood correctly. It's buffalo handler, isn't it? I took inspiration from the other handlers.

Reply:
I was just wondering if it was possible to do so.

Reply:
I think we cant since a buffalo.Handler is a `func(buffalo.Context) error`